pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -0,0 +1,300 @@
+
+		cachesToSync: []cache.InformerSynced{
+			operatorConfigClient.Informer().HasSynced,
+			operatorInformers.Operator().V1().Etcds().Informer().HasSynced,
+			operatorInformers.Operator().V1().KubeAPIServers().Informer().HasSynced,
+		},
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BootstrapTeardownController"),
+		eventRecorder: eventRecorder.WithComponentSuffix("cluster-member-controller"),
+	}
+
+	operatorInformers.Operator().V1().KubeAPIServers().Informer().AddEventHandler(c.eventHandler())
+	operatorInformers.Operator().V1().Etcds().Informer().AddEventHandler(c.eventHandler())
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *BootstrapTeardownController) sync() error {
+	err := c.removeBootstrap()
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorConfigClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "BootstrapTeardownDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Error",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("BootstrapTeardownErrorUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorConfigClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "BootstrapTeardownDegraded",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}),
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    ConditionBootstrapRemoved,
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "BootstrapNodeRemoved",
+			Message: "Etcd operator has scaled",
+		}))
+	return updateErr
+}
+
+func (c *BootstrapTeardownController) removeBootstrap() error {
+	currentEtcdOperator, err := c.etcdOperatorLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	if currentEtcdOperator.Spec.ManagementState != operatorv1.Managed {
+		return nil
+	}
+
+	etcdReady, err := c.isEtcdAvailable(currentEtcdOperator)
+	if err != nil {
+		return err
+	}
+
+	if !etcdReady {
+		klog.Infof("Still waiting for the cluster-etcd-operator to bootstrap...")
+		return fmt.Errorf("waiting for cluster-etcd-operator to bootstrap")
+	}
+
+	kubeAPIServer, err := c.kubeAPIServerLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	kasReady := c.isKASReady(kubeAPIServer, c.configMapsGetter)
+	if !kasReady {
+		klog.Infof("Still waiting for the kube-apiserver to be ready...")
+		return fmt.Errorf("waiting for kube-apiserver to be ready")
+	}
+
+	if !c.clusterMemberShipController.IsMember("etcd-bootstrap") {
+		return nil
+	}
+
+	c.eventRecorder.Event("BootstrapTeardownController", "safe to remove bootstrap")
+	if err := c.clusterMemberShipController.RemoveBootstrap(); err != nil {
+		return err
+	}
+
+	// set bootstrap removed condition
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorConfigClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
```

Comment:
this is fine to improve latency, but doing this here is *not* sufficient for level driven behavior. This is a reaction on an edge.

Reply:
> this is fine to improve latency, but doing this here is _not_ sufficient for level driven behavior. This is a reaction on an edge.

actually, remove this. It's likely that you'll produce a bug doing it this way.